crates/engine/tree/src/tree/mod.rs

```diff
@@ -342,9 +366,38 @@ where
                     }
                 }
             }
+
+            if self.should_persist() {
```

Comment:
Simplest possible hardcoded hook, should we use instead something similar to https://github.com/paradigmxyz/reth/tree/main/crates/consensus/beacon/src/engine/hooks ?